coupled_laplace_problem/coupled_laplace_problem.cc

```diff
@@ -0,0 +1,640 @@
+  const std::string mesh_name;
+  const std::string read_data_name;
+
+  // These IDs are filled by preCICE during the initialization. We set a default
+  // value of -1 in order to detect potential errors more easily.
+  int mesh_id           = -1;
+  int read_data_id      = -1;
+  int n_interface_nodes = -1;
+
+  // Dof IndexSet, containing relevant coupling dof indices at the coupling
+  // boundary
+  IndexSet coupling_dofs;
+
+  // Data containers which are passed to preCICE in an appropriate preCICE
+  // specific format
+  std::vector<int>    interface_nodes_ids;
+  std::vector<double> read_data;
+
+  // Function to transform the obtained data from preCICE into an appropriate
+  // map for Dirichlet boundary conditions
+  void
+  format_precice_to_deal(
+    std::map<types::global_dof_index, double> &boundary_data) const;
+};
+
+
+
+// In the constructor of the Adapter class, we set up the preCICE
+// Solverinterface. Here, we need to tell preCICE our name as participant of the
+// simulation and the name of the preCICE-configuration file. Both have already
+// been specified in the CouplingParameter class above. Thus, we pass the class
+// directly to the constructor and read out all relevant information. As a
+// second parameter, we need to specify the boundary ID of our triangulation,
+// which is associated with the coupling interface.
+template <int dim, typename ParameterClass>
+Adapter<dim, ParameterClass>::Adapter(
+  const ParameterClass &   parameters,
+  const types::boundary_id deal_boundary_interface_id)
+  : precice(parameters.participant_name,
+            parameters.config_file,
+            Utilities::MPI::this_mpi_process(MPI_COMM_WORLD),
+            Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD))
+  , deal_boundary_interface_id(deal_boundary_interface_id)
+  , mesh_name(parameters.mesh_name)
+  , read_data_name(parameters.read_data_name)
+{}
+
+
+
+// This function initializes preCICE (e.g. establishes communication channels
+// and allocates memory) and passes all relevant data to preCICE. For surface
+// coupling, relevant data is especially the location of the data points at the
+// assoociated interface(s). The `boundary_data` is an empty map, which is
+// filled by preCICE, i.e., the information of our other participant. Throughout
+// the system assembly, the map can directly be used in order to apply the
+// Dirichlet boundary conditions in the linear system. preCICE returns during
+// the initialization the maximum admissible time-step size.
+template <int dim, typename ParameterClass>
+double
+Adapter<dim, ParameterClass>::initialize(
+  const DoFHandler<dim> &                    dof_handler,
+  std::map<types::global_dof_index, double> &boundary_data)
+{
+  Assert(dim > 1, ExcNotImplemented());
+  Assert(dim == precice.getDimensions(),
+         ExcDimensionMismatch(dim, precice.getDimensions()));
+
+  // In a first step, we get precice specific IDs from precice and store them in
```

Comment:
```suggestion
  // In a first step, we get preCICE specific IDs from preCICE and store them in
```
Please check everywhere for consistency.